### PR TITLE
Patch: Bug fixes and config hardening

### PR DESCRIPTION
**Pull request by MehdiZare**
## Summary

- Fix `RateLimitConfig.from_env()` crash when env vars are set to zero or negative values (e.g. `FMP_DAILY_LIMIT=0`) — now gracefully falls back to defaults instead of crashing Pydantic validation (#76)
- Harden `ClientConfig.from_env()` with the same pattern for `FMP_TIMEOUT` and `FMP_MAX_RETRIES`
- Document undocumented rate-limit env vars in CLAUDE.md

## Test plan

- [x] All 1114 unit tests pass
- [x] New edge-case tests for zero/negative env values
- [x] Lint and formatting clean
- [x] Manual smoke test confirms defaults are used instead of crashing

Closes #76

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Automated release PR creation from development to main branch
  * Automatic synchronization of changes from main back to development
  * Rebase reminders for PRs behind the development branch

* **Improvements**
  * Enhanced PR validation workflow to support release branch types

* **Documentation**
  * Added environment variable configuration reference for rate limiting and timeout settings

* **Tests**
  * Improved test coverage for configuration edge cases and defaults

### CODE REVIEW

File: .github/workflows/sync-main-to-dev.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            - Prevents future merge conflicts
+
+            ### Action Required
+            - Review the changes
+            - Resolve any conflicts if present
+            - **Use Squash and Merge** to preserve linear history on `dev`
+          labels: |
+            automated
+            sync
+          delete-branch: false
```
[the remark this answers is not in the document]
Comment (MehdiZare) [reply .github/workflows/sync-main-to-dev.yml]: **Addressed** — Updated to latest major version.

- Updated `peter-evans/create-pull-request` from `@v6` to `@v8` in both `sync-main-to-dev.yml` and `release-pr.yml`.
- Keeping version tag pinning (not SHA pinning) to stay consistent with the repo's existing tag-pinning strategy and Dependabot config.
